src/vector_index.c

```diff
@@ -290,12 +288,48 @@ int VecSim_RdbLoad(RedisModuleIO *rdb, VecSimParams *vecsimParams) {
   return REDISMODULE_ERR;
 }
 
-int VecSimResolveCode_to_QueryErrorCode(int code) {
+VecSimResolveCode VecSim_ResolveQueryParams(VecSimIndex *index, VecSimRawParam *params, size_t params_len,
+                          VecSimQueryParams *qParams, bool hybrid, QueryError *status) {
+
+  VecSimResolveCode code = VecSimIndex_ResolveParams(index, params, params_len, qParams, hybrid);
+  if (code == VecSim_OK) {
+    return code;
+  }
+
+  const char *error_msg = "";
   switch (code) {
-    case VecSim_OK: return QUERY_OK;
-    case VecSimParamResolverErr_AlreadySet: return QUERY_EDUPFIELD;
-    case VecSimParamResolverErr_UnknownParam: return QUERY_ENOOPTION;
-    case VecSimParamResolverErr_BadValue: return QUERY_EBADATTR;
+    case VecSimParamResolverErr_AlreadySet: {
+      error_msg = QueryError_Strerror(QUERY_EDUPFIELD);
+      break;
+    }
+    case VecSimParamResolverErr_UnknownParam: {
+      error_msg = QueryError_Strerror(QUERY_ENOOPTION);
+      break;
+    }
+    case VecSimParamResolverErr_BadValue: {
+      error_msg = QueryError_Strerror(QUERY_EBADVECSIMATTR);
+      break;
+    }
+    case VecSimParamResolverErr_InvalidPolicy_NHybrid: {
+      error_msg = "hybrid query attributes were sent for a non-hybrid VSS query";
```

Comment:
Because for some error codes, there were already suitable messages, while other don't.  Now that I think of it, I will add the new messages to the place where the old messages are.